lib/agent0/agent0/utilities/predict.py

```diff
@@ -0,0 +1,234 @@
+"""Predict the outcome of trades."""
+
+from __future__ import annotations
+
+import logging
+from copy import deepcopy
+from typing import NamedTuple
+
+from ethpy.hyperdrive.interface.read_interface import HyperdriveReadInterface
+from ethpy.hyperdrive.state import PoolState
+from fixedpointmath import FixedPoint
+
+# need to cover a wide variety of cases
+# pylint: disable=too-many-arguments
+# being very explicit
+# pylint: disable=too-many-locals
+
+YEAR_IN_SECONDS = 31_536_000
+BLOCKS_IN_YEAR = YEAR_IN_SECONDS / 12
+
+Deltas = NamedTuple(
+    "Deltas",
+    [
+        ("base", FixedPoint),
+        ("bonds", FixedPoint),
+        ("shares", FixedPoint),
+    ],
+)
+TradeDeltas = NamedTuple(
+    "TradeDeltas",
+    [
+        ("user", Deltas),
+        ("pool", Deltas),
+        ("fee", Deltas),
+        ("governance", Deltas),
+    ],
+)
+
+
+def predict_long(
+    hyperdrive_interface: HyperdriveReadInterface,
+    pool_state: PoolState | None = None,
+    base: FixedPoint | None = None,
+    bonds: FixedPoint | None = None,
+    for_pool: bool = False,
+    verbose: bool = False,
+) -> TradeDeltas:
+    """Predict the outcome of a long trade.
+
+    Arguments
+    ---------
+    hyperdrive_interface: HyperdriveReadInterface
+        Hyperdrive interface.
+    pool_state: PoolState, optional
+        The state of the pool, which includes block details, pool config, and pool info.
+        If not given, use the current pool state.
+    base: FixedPoint, optional
+        The size of the long to open, in base. If not given, converted from bonds.
+    bonds: FixedPoint, optional
+        The size of the long to open, in bonds.
+    for_pool: bool
```

Comment:
You have `for_pool = True` in the only use case (`lpandarb.py`) and `for_pool=False` in all of the tests in `predict_trade_test.py`. To simplify the code _and_ have the test more accurately reflect the usage, I recommend you remove this variable entirely, only include the `for_pool=True` branch, and modify the test to assume this is the case.

Reply:
created tracking issue, left comment in PR